target/linux/ramips/base-files/lib/ramips.sh

```diff
@@ -85,6 +85,9 @@ ramips_board_detect() {
 	*"C108")
 		name="c108"
 		;;
+	*"C2")
```

Comment:
drop this and use `tplink,c2` as userspace boardname (from the dts compatible) everywhere.

Reply:
Sorry but I don't understand how this should work. Please explain in more detail how this should work

Reply:
In these other files, `tplink,c2` is wanted, not just `c2`:
- target/linux/ramips/base-files/etc/board.d/01_leds
- target/linux/ramips/base-files/etc/board.d/02_network
- target/linux/ramips/base-files/etc/diag.sh
- target/linux/ramips/base-files/lib/upgrade/platform.sh

You may need to move your code to fix ordering.

Example https://github.com/openwrt/openwrt/commit/5bdd238b634a34b038fbea9046893de29a77c000